### PR TITLE
fix 0.17.0 release

### PR DESCRIPTION
**Pull request by abdulajet**
### Contributing checklist
- [x] Code must follow existing styling conventions
- [x] Added a descriptive commit message

`allEvents` was checking the wrong key on native events
fixed iOS opentok version to 2.18.1.
moved version to 0.17.0

tested with basic video chat and signalling demo on ios and android

### CODE REVIEW

File: src/OT.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ const setNativeEvents = (events) => {
   const eventNames = Object.keys(events);
   OT.setNativeEvents(eventNames);
   each(events, (eventHandler, eventType) => {
-    const allEvents = Object.keys(nativeEvents['_nativeModule']);
+    const allEvents = Object.keys(nativeEvents['_subscriber']['_subscriptionsForType']);
```
Comment (ggoldens) [inline src/OT.js:35]: Could you please explain a bit more what was the issue with this and how this is fixing it? Thanks!

**Review by ggoldens — COMMENTED** [review]
Please, check my comment. Thanks!

File: src/OT.js (hunk 1 above)
Comment (abdulajet) [reply src/OT.js:35]: The `nativeEvents` object has two sub-objects on iOS `_nativeModule` and `_subscriber`, but only `_subscriber` on android. 

Looking at the events in `eventsNames` and the contents of `_nativeModule` there weren't any matches, whereas the array on '_subscriber' -> '_subscriptionsForType' had the events I was looking for. 

With hindsight, it makes sense as the JavaScript is "subscribing" to events from the native side.

Comment (mbouxin) [reply src/OT.js:35]: It fixes #477 using inner keys of NativeEventEmitter class. 

A more robust way would be to use the new listenerCount method

Comment (abdulajet) [reply src/OT.js:35]: That method seems to only return a number of events? My understanding of the original code of this function was to get a list of events and compare if the event has already been added.

Comment (mbouxin) [reply src/OT.js:35]: Yes.. and if a listener has been added than listenerCount is greater than 0. I opened a PR #498 to be clearer

Comment (abdulajet) [reply src/OT.js]: tested it on older versions as well and work well, couldn't cherry-pick your commit unfortunately but thanks

**Review by ggoldens — APPROVED** [review]
LGTM

File: opentok-react-native.podspec [hunk 2]
```diff
@@ -17,5 +17,5 @@ Pod::Spec.new do |s|
   s.source_files  = "ios/**/*.{h,m,swift}"
 
   s.dependency 'React'
-  s.dependency 'OpenTok'
+  s.dependency 'OpenTok','2.18.1'
```
Comment (enricop89) [inline opentok-react-native.podspec:20]: Shall we update to the latest `2.19.1`?

Comment (abdulajet) [reply opentok-react-native.podspec:20]: not as part of this release I don't think

**Review by enricop89 — APPROVED** [review]
LGTM

**Pull request merged**